ChopChopTests/Models/Ingredients/QuantityTests.swift

```diff
@@ -0,0 +1,287 @@
+        let rightCount: Quantity = .count(5)
+        XCTAssertNoThrow(try leftCount -= rightCount)
+        XCTAssertEqual(leftCount, .count(6 - 5))
+
+        var leftMass: Quantity = .mass(0.4)
+        let rightMass: Quantity = .mass(0.3)
+        XCTAssertNoThrow(try leftMass -= rightMass)
+        XCTAssertEqual(leftMass, .mass(0.4 - 0.3))
+
+        var leftVolume: Quantity = .volume(0.2)
+        let rightVolume: Quantity = .volume(0.1)
+        XCTAssertNoThrow(try leftVolume -= rightVolume)
+        XCTAssertEqual(leftVolume, .volume(0.2 - 0.1))
+    }
+
+    func testMinusEquals_differentQuantityType_throwsError() {
+        var left: Quantity = .count(1)
+        let right: Quantity = .volume(0.5)
+
+        XCTAssertThrowsError(try left -= right)
+    }
+
+    func testMinusEquals_negativeResult_throwsError() {
+        var left: Quantity = .volume(0.5)
+        let right: Quantity = .volume(1)
+
+        XCTAssertThrowsError(try left -= right)
+    }
+
+    func testTimesEquals_validResult_success() {
+        var testCount: Quantity = .count(7)
+        var testMass: Quantity = .mass(0.8)
+        var testVolume: Quantity = .volume(0.9)
+        let factor: Double = 1.5
+
+        XCTAssertNoThrow(try testCount *= factor)
+        XCTAssertNoThrow(try testMass *= factor)
+        XCTAssertNoThrow(try testVolume *= factor)
+        XCTAssertEqual(testCount, .count(7 * factor))
+        XCTAssertEqual(testMass, .mass(0.8 * factor))
+        XCTAssertEqual(testVolume, .volume(0.9 * factor))
+    }
+
+    func testTimesEquals_negativeResult_throwsError() {
+        var quantity: Quantity = .mass(1)
+        let factor: Double = -0.3
+
+        XCTAssertThrowsError(try quantity *= factor)
+    }
+
+    func testDividesEquals_validResult_success() {
+        var testCount: Quantity = .count(1.1)
+        var testMass: Quantity = .mass(1.2)
+        var testVolume: Quantity = .volume(1.3)
+        let factor: Double = 4
+
+        XCTAssertNoThrow(try testCount /= factor)
+        XCTAssertNoThrow(try testMass /= factor)
+        XCTAssertNoThrow(try testVolume /= factor)
+        XCTAssertEqual(testCount, .count(1.1 / factor))
+        XCTAssertEqual(testMass, .mass(1.2 / factor))
+        XCTAssertEqual(testVolume, .volume(1.3 / factor))
+    }
+
+    func testDividesEquals_negativeResult_throwsError() {
+        var quantity: Quantity = .volume(1.4)
+        let factor: Double = -2
+
+        XCTAssertThrowsError(try quantity /= factor)
+    }
+
+    func testDividesEquals_zeroDivisor_throwsError() {
+        var quantity: Quantity = .count(1.5)
+        let factor: Double = 0
+
+        XCTAssertThrowsError(try quantity /= factor)
+    }
+}
+
+// MARK: - Comparable
+extension QuantityTests {
+    func testLessThan_sameQuantityType_success() {
+        let leftCount: Quantity = .count(1)
+        let rightCount: Quantity = .count(2)
+        XCTAssertTrue(try leftCount < rightCount)
+        XCTAssertFalse(try rightCount < leftCount)
```

Comment:
Any reason why relevant methods like `XCTAssertLessThan` are not used?

Reply:
Thanks for the suggestion!